mos_tests/ironic/conftest.py

```diff
@@ -93,3 +95,92 @@ def baremetal_node(env_name, suffix):
     node.erase()
     disk.volume.erase()
     disk.delete()
+
+
+@pytest.yield_fixture
+def keypair(os_conn):
+    keypair = os_conn.create_key(key_name='ironic-key')
+    yield keypair
+    os_conn.delete_key(key_name=keypair.name)
+
+
+@pytest.yield_fixture
+def flavor(baremetal_node, os_conn):
+    flavor = os_conn.nova.flavors.create(name=baremetal_node.name,
+                                         ram=baremetal_node.memory,
+                                         vcpus=baremetal_node.vcpu,
+                                         disk=settings.IRONIC_DISK_GB)
+
+    yield flavor
+    flavor.delete()
+
+
+@pytest.yield_fixture(scope='module')
+def image_file():
+    image_file = common.gen_temp_file(prefix='image', suffix='img')
+    yield image_file
+    image_file.unlink(image_file.name)
+
+
+@pytest.yield_fixture
+def ubuntu_image(os_conn, image_file):
+    image = os_conn.glance.images.create(
+        name='ironic_trusty',
+        disk_format='raw',
+        container_format='bare',
+        hypervisor_type='baremetal',
+        cpu_arch='x86_64',
+        fuel_disk_info=json.dumps(settings.IRONIC_GLANCE_DISK_INFO))
+
+    if not image_file.file.closed:
+        src = urllib.request.urlopen(settings.IRONIC_IMAGE_URL)
+        with tarfile.open(fileobj=src, mode='r|gz') as tar:
+            img = tar.extractfile(tar.firstmember)
+            while True:
+                data = img.read(1024)
+                if not data:
+                    break
+                image_file.file.write(data)
+            image_file.file.close()
+        src.close()
+    with open(image_file.name) as f:
+        os_conn.glance.images.upload(image.id, f)
+
+    yield image
+    os_conn.glance.images.delete(image.id)
+
+
+@pytest.yield_fixture
+def ironic_node(baremetal_node, os_conn, ironic, server_ssh_credentials):
+
+    def get_image(name):
+        return os_conn.nova.images.find(name=name)
+
+    driver_info = {
+        'ssh_address': server_ssh_credentials['ip'],
```

Comment:
Also good to implement driver_info settings set for IPMI driver in case our testing lab will have some real hardware nodes with IPMI interface.